site-src/geps/gep-735.md

```diff
@@ -0,0 +1,170 @@
+# GEP-735: TCP and UDP addresses matching
+
+* Issue: [#735](https://github.com/kubernetes-sigs/gateway-api/issues/735)
+* Status: Provisional
+
+## TLDR
+
+Spec for matching source and destination addresses on L4 APIs.
+
+## Goals
+
+- add matching rules for address to `TCPRoute`
+- add matching rules for address to `UDPRoute`
+- intentionally avoid type definitions that would make it hard to expand later
+
+## Non-Goals
+
+- define rules for port matching
+
+## Introduction
+
+While `TCPRoute` and `UDPRoute` currently support custom matching extensions,
+there is desire among the community to include some "fundamental" matching
+options in the spec that cover the most common requirements. In this GEP we
+request address matching for these APIs in order to support a standard
+for some of the commonplace setups of gateway implementations. Matching is
+intended to be covered for both _source_ and _destination_ to enable a finer
+level of tuning options for L4 traffic routing at a level below the `Gateway`.
+
+## API
+
+The API changes include the following new types:
+
+- `AddressMatch` to indicate the IP for address matching
+- `TrafficMatches` to configure matching according to network address
+
+These types enable the address matching required, with some active
+considerations about how to leave these open ended for later expansion.
+
+### AddressMatch Type
+
+A new `AddressMatch` type provides the targeting mechanism for match inclusion
+of a given network address:
+
+```go
+type AddressMatch struct {
+	// Type of the address, either IPAddress or NamedAddress.
+	//
+	// If NamedAddress is used this is a custom and specific value for each
+	// implementation to handle (and add validation for) according to their
+	// own needs.
+	//
+	// For IPAddress the implementor may expect either IPv4 or IPv6.
+	//
+	// +optional
+	// +kubebuilder:validation:Enum=IPAddress;NamedAddress
+	// +kubebuilder:default=IPAddress
+	Type *AddressType `json:"type,omitempty"`
```

Comment:
Please add conformance levels for these features.

Reply:
Now added in c25e4f68885700d948c10c5534c7e52bd2dbf297 :+1: